controller/destination/server.go

```diff
@@ -104,6 +127,32 @@ func (s *server) Get(dest *common.Destination, stream pb.Destination_GetServer)
 	return nil
 }
 
+func (s *server) localKubernetesServiceIdFromDNSName(host string) (*string, error) {
+	hostLabels := strings.Split(host, ".")
+
+	// Verify that `host` ends with .svc.$zone.
+	if len(hostLabels) <= 1 + len(s.k8sDNSZoneLabels) {
+		return nil, nil
+	}
+	n := len(hostLabels) - len(s.k8sDNSZoneLabels)
+	if !reflect.DeepEqual(hostLabels[n:], s.k8sDNSZoneLabels) {
```

Comment:
feels like more idiomatic golang to use a for loop to iterate through this subslice and check string equality than to use reflect